R/install_torsten.R

```diff
@@ -0,0 +1,197 @@
+#' @param quiet (logical) Should the verbose output
+#'   from the system processes be suppressed when building the Torsten binaries?
+#'   The default is `FALSE`.
+#' @param version (string) The Torsten release version to install. The default
+#'   is `NULL`, which downloads the latest stable release from
+#'   <https://github.com/metrumresearchgroup/Torsten/releases>.
+#' @param release_url (string) The URL for the specific Torsten release or
+#'   release candidate to install. See <https://github.com/metrumresearchgroup/Torsten/releases>.
+#'   The URL should point to the tarball (`.tar.gz.` file) itself, e.g.,
+#'   `release_url="https://github.com/metrumresearchgroup/Torsten/archive/refs/tags/torsten_v0.90.0.tar.gz"`.
+#'   If both `version` and `release_url` are specified then `version` will be used.
+#' @param cpp_options (list) Any makefile flags/variables to be written to
+#'   the `make/local` file. For example, `list("CXX" = "clang++")` will force
+#'   the use of clang for compilation.
+#' @param check_toolchain (logical) Should `install_torsten()` attempt to check
+#'   that the required toolchain is installed and properly configured. The
+#'   default is `TRUE`.
+#'
+#' @examples
+#' \dontrun{
+#' # install_torsten(cores = 4)
+#'
+#' cpp_options <- list(
+#'   "CXX" = "clang++",
+#'   "CXXFLAGS+= -march=native",
+#'   PRECOMPILED_HEADERS = TRUE
+#' )
+#' # install_torsten(cores = 4, cpp_options = cpp_options)
+#' }
+#'
+
+#'## install_torsten version without unexported dependencies
+
+latest_torsten_release <- function() {
+  require(jsonlite)
+  dest_file <- tempfile(pattern = "releases-", fileext = ".json")
+  download_url <- "https://api.github.com/repos/metrumresearchgroup/Torsten/releases/latest"
+  release_list_downloaded <- utils::download.file(url = download_url,
+                                                  destfile = dest_file,
+                                                  quiet = TRUE)
+  if (release_list_downloaded != 0) {
+    stop("GitHub download of release list failed.", call. = FALSE)
+  }
+  release <- jsonlite::read_json(dest_file)
+  release$tag_name
+}
+
+get_torsten_release_list <- function() {
+  require(jsonlite)
+  dest_file <- tempfile(pattern = "releases-", fileext = ".json")
+  download_url <- "https://api.github.com/repos/metrumresearchgroup/Torsten/releases"
+  release_list_downloaded <- utils::download.file(url = download_url,
+                                                  destfile = dest_file,
+                                                  quiet = TRUE)
+  if (release_list_downloaded != 0) {
+    stop("GitHub download of release list failed.", call. = FALSE)
+  }
+  release <- jsonlite::read_json(dest_file)
+  sapply(release,
+         function(x){
+           x$tag_name
+         })
+}
+
+install_torsten <- function(dir = NULL,
+                            cores = getOption("mc.cores", 2),
+                            quiet = FALSE,
+                            timeout = 1200,
```

Comment:
A description for `timeout` is missing from the documentation.

Side remark: a lot of the arguments are shared with `cmdstanr::install_cmdstan`, so I was going to mention that you could use `@inheritParams cmdstanr::install_cmdstan` to avoid needing to repeat a lot of the descriptions. However, I see that most of the descriptions won't quite work as is (e.g., "Torsten" is used instead of "cmdstanr").

It's not all or none, so you could still use `@inheritParams` for the ones that are identical (looks like `cpp_options` and `cores`), but it's fine as is.

Reply:
`timeout` description added